java-spiffe-core/src/main/java/spiffe/bundle/x509bundle/X509BundleSet.java

```diff
@@ -0,0 +1,70 @@
+package spiffe.bundle.x509bundle;
+
+import lombok.NonNull;
+import lombok.Value;
+import lombok.val;
+import spiffe.bundle.BundleSource;
+import spiffe.exception.BundleNotFoundException;
+import spiffe.spiffeid.TrustDomain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A <code>X509BundleSet</code> represents a set of X.509 bundles keyed by trust domain.
+ */
+@Value
+public class X509BundleSet implements BundleSource<X509Bundle> {
+
+    ConcurrentHashMap<TrustDomain, X509Bundle> bundles;
+
+    private X509BundleSet(final Map<TrustDomain, X509Bundle> bundles) {
+        this.bundles = new ConcurrentHashMap<>(bundles);
+    }
+
+    /**
+     * Creates a new X.509 bundle set from a list of X.509 bundles.
+     *
+     * @param bundles a list of {@link X509Bundle}
+     * @return a {@link X509BundleSet} initialized with the list of bundles
+     */
+    public static X509BundleSet of(@NonNull final List<X509Bundle> bundles) {
+        Map<TrustDomain, X509Bundle> bundleMap = new HashMap<>();
+        for (X509Bundle bundle : bundles) {
+            bundleMap.put(bundle.getTrustDomain(), bundle);
+        }
+        return new X509BundleSet(bundleMap);
+    }
+
+    /**
+     * Adds a bundle to this Set, if the trust domain already exists,
+     * replaces the bundle.
+     *
+     * @param x509Bundle a {@link X509Bundle}
+     */
+    public void add(@NonNull X509Bundle x509Bundle){
```

Comment:
I think `put()` is probably a more appropriate name since this can replace existing entries. `add()` to me implies that the method would not modify/remove existing elements in the `X509BundleSet`.

Reply:
`Map<K, V>` also uses the name `put()` for this operation, so I think that name more closely aligns with Java developers' expectations.

Reply:
I think you are right, good catch.